RestaurentManagement/src/main/java/com/restapi/service/MenuService.java

```
@@ -91,10 +68,17 @@ public Menu updateMenu(UUID id, Menu menu) {
     }
 
     public void deleteMenu(UUID id) {
-        if (!menuRepository.existsById(id)) {
-            ErrorDetail errorDetail = createNotFoundErrorDetail(id);
-            throw new ResourceException(errorDetail);
+        try {
+            if (!menuRepository.existsById(id)) {
+                throw new EmptyResultDataAccessException(1);
+            }
+            menuRepository.deleteById(id);
+        } catch (EmptyResultDataAccessException e) {
```

Just need to check the existing ID and throw an exception if not found